probes/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.12 AS build-env
+
+ARG gopkg=github.com/mm4tt/k8s-util/experimental/probes
+
+ADD ["cmd", "/go/src/$gopkg/cmd"]
+ADD ["pkg", "/go/src/$gopkg/pkg"]
+ADD ["go.mod", "/go/src/$gopkg"]
+ADD ["go.sum", "/go/src/$gopkg"]
+
+ENV GO111MODULE on
+WORKDIR /go/src/$gopkg
+RUN CGO_ENABLED=0 go build -o /workspace/probes ./cmd
+
+FROM golang:1.12-alpine
+WORKDIR /workspace
+COPY --from=build-env /workspace/probes .
+ENTRYPOINT ["/workspace/probes"]
```

Comment:
I'm not huge fan of entrypoint - I personally prefer explicitly running command when starting a container rather than relying on some default for this. Can we remove this?

Reply:
Every probe is different and requires different arguments, that's why I set only the image **entrypoint** (which is exactly the same for every probe) and didn't set the image **cmd** (which will be different).
See https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/

Currently, if you execute 
```
docker run gcr.io/k8s-testimages/probes:v0.0.1
```
You'll get
```
F0408 13:52:18.015111       1 main.go:63] --mode not set!
```

Then you can execute the same with `--help` or with `-mode=XXX` to see what probes are supported and what other flags are required.
If I removed the entrypoint then executing the docker run would produce nothing. I'm quite sure it would be confusing for some people (e.g. I'd find it confusing if I didn't know the set-up :))

Reply:
I though that if default doesn't work for you (which is the case here, because you need to pass flags), then you need to set the command too - aparently it's enough to pass args.

So I'm fine with this - thanks!